.github/workflows/continuous-integration-workflow.yml

```diff
@@ -40,11 +40,26 @@ jobs:
           repository:  todo-app
           tag: main
 
-  refresh-azure-image:
-    name: Refresh Azure Image
+  update-infrastructure:
+    name: Terraform changes
     needs: build-and-push-docker-image
+    if: ${{ github.ref == 'refs/heads/main' }}
+    env:
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+      ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}

```

Comment:
If you do change `SECRET_KEY` to be a variable, just setting it here like:
```yml
      TF_VAR_secret_key: ${{ secrets.SECRET_KEY }}
```
Will match it up automatically to a variable in your Terraform called `secret_key`